PassleSync.Website/App_Plugins/PassleSync/backoffice/passleSync/components/organisms/authorsTable/authorsTable.controller.js

```diff
@@ -1,271 +1,31 @@
-                syncProm = passleAuthorsResource.syncMany(shortcodes);
-            }
-
-            syncProm.then((response) => {
-                vm.authors.forEach((author, ii) => {
-                    let matchingAuthors = response.Authors.filter(x => x.Shortcode === author.shortcode);
-                    if (matchingAuthors.length > 0) {
-                        vm.authors[ii] = getAuthorDataObject(matchingAuthors[0]);
-                    }
-                });
-
-                vm.syncedCount = vm.authors.filter((author) => author.synced).length;
-                vm.unsyncedCount = vm.authors.length - vm.syncedCount;
-                vm.isSelectedAll = false;
-                vm.selectedCount = 0;
-
-                UpdatePagination();
-                UpdateItemsOnShow();
-
-                vm.isUpdating = false;
-
-                notificationsService.success("Success", "Authors have been synced");
-
-                syncTree();
-            }, (error) => {
-                console.error(error);
-                notificationsService.error("Error", error);
-
-                vm.isUpdating = false;
-            });
-        }
-
-        vm.delete = function () {
-            vm.isUpdating = true;
-
-            let deleteProm;
-            let shortcodes;
-
-            if (vm.selectedCount == 0) {
-                deleteProm = passleAuthorsResource.deleteAll();
-            } else if (vm.selectedCount == 1) {
-                shortcodes = vm.authors.filter((author) => author.selected).map((author) => author.shortcode);
-                deleteProm = passleAuthorsResource.deleteOne(shortcodes);
-            } else {
-                shortcodes = vm.authors.filter((author) => author.selected).map((author) => author.shortcode);
-                deleteProm = passleAuthorsResource.deleteMany(shortcodes);
-            }
-
-            deleteProm.then((response) => {
-                vm.authors.forEach((author, ii) => {
-                    let matchingAuthors = response.Authors.filter(x => x.Shortcode === author.shortcode);
-                    if (matchingAuthors.length > 0) {
-                        vm.authors[ii] = getOverriddenAuthorDataObject(matchingAuthors[0]);
-                    }
+        vm.entityInfo = {
+            nameSingle: "Author",
+            namePlural: "Authors",
+            getRowData: function getRowData(author, deletedOverride = false) {
+                // Convert the returned model to the format the table needs
+                // The override is used when deleting data:
+                // - we return the last state of the content before it was deleted, so we need to override the synced and url values
+
+                return Object.assign({}, {
+                    "name": author.Name,
+                    "role": author.RoleInfo,
+                    "editPath": (author.Synced && !deletedOverride) ? ("/content/content/edit/" + author.Id) : author.ProfileUrl,
```

Comment:
Similar to the postsTable, I think it would make more sense to always return the `author.ProfileUrl`